sources/packages/backend/apps/api/src/route-controllers/education-program-offering/education-program-offering.students.controller.ts

```diff
@@ -93,4 +104,44 @@ export class EducationProgramOfferingStudentsController extends BaseController {
       studyEndDate: offering.studyEndDate,
     };
   }
+
+  /**
+   * Gets the offering simplified details, not including, for instance,
+   * validations, approvals and extensive data.
+   * Useful to have an overview of the offering details, for instance,
+   * when the user needs need to have quick access to data in order to
+   * support operations like confirmation of enrolment or scholastic
+   * standing requests or offering change request.
+   * @param offeringId offering.
+   * @param purpose indicating the purpose to allow for the appropriate authorization flow to be used.
+   * @param studentUserToken student user token for authorization.
+   * @returns offering details.
+   */
+  @ApiUnauthorizedResponse({
+    description: "The student is not authorized for the provided offering.",
+  })
+  @Get("offering/:offeringId/summary-details")
+  async getOfferingSummaryDetailsById(
+    @Param("offeringId", ParseIntPipe) offeringId: number,
+    @Query("purpose", new ParseEnumQueryPipe(ApplicationOfferingPurpose))
+    purpose: ApplicationOfferingPurpose,
+    @UserToken() studentUserToken: StudentUserToken,
+  ): Promise<EducationProgramOfferingSummaryViewAPIOutDTO> {
+    if (
+      purpose === ApplicationOfferingPurpose.OfferingChange &&
+      (await this.educationProgramOfferingControllerService.validateApplicationOfferingForStudent(
+        offeringId,
+        studentUserToken.studentId,
+      ))
+    )
+      return this.educationProgramOfferingControllerService.getOfferingById(
+        offeringId,
+      );
+    throw new UnauthorizedException(
```

Comment:
I am sorry, Iam not following this error logic, was your intention to add it in a try-catch?

Reply:
Fixed it.